tools/src/format.rs

```diff
@@ -323,6 +323,7 @@ pub fn format_file_data(
     cfg: &config::Config,
     tree_name: &str,
     panel: &[PanelSection],
+    info_boxes: Vec<InfoBox>,
```

Comment:
There was an ownership issue related to the `F`s that I failed to document.  I've pushed separate commit https://github.com/mozsearch/mozsearch/pull/357/commits/b961218b2f9e5ee0d594680ff997d7ad79cb6d89 that adds a clone mechanism, but I'm open to other fixes.  The only real opinion I have about the logic is it's nice to defer the formatting to as late as possible so it's nice for InfoBox to take a formatting object.  (Otherwise the indentation would be messed up unless we treat F to do appropriate replacements on newlines, but that's hacky/potentially breakage-prone for script tags with inline contents.)

Reply:
I guess if there were an enum variant similar to `Indent` but took a reference then we wouldn't need the clone. But yeah we can leave that for a future followup

Reply:
Going to take you up on the deferral for now.  For future work, I think it's worth considering whether we could move to an existing third-party crate that might look more like React's JSX.  That is, a means of using macros to more directly embed something that looks like HTML and get some type-checking that the HTML is valid rather than a bunch of manual string juggling with format! calls throughout.

Unfortunately, it looks like the existing options are either unmaintained (ex: https://github.com/bodil/typed-html which seems nice, https://github.com/utkarshkukreti/markup.rs also seems interesting) or part of a massive transpiling-to-WASM web framework (ex: [yew's html! macro](https://yew.rs/docs/concepts/html)).